### ISSUE TITLE
PSA: The Clara 2E's touch panel sucks

### ISSUE THREAD

**Issue by NiLuJe** [koreader/koreader#12634]
* KOReader version: any
* Device: Kobo Clara 2E

#### Issue

Story time: given the semi-death of my Forma, I switched my on-the-go device from a Forma to a Clara 2E.
I barely ever read PDFs, so everything was fine and dandy until then...
But, when reading PDFs, and playing with the Classic or Turbo scrolling, I noticed all kinds of weirdness, like taps/hold/extra pans going through *during* my scrolling...

Being somewhat intimately familiar with the input & gesture detector framework, it looked to me like spurious contact lifts (and, more importantly, I didn't remember anything of the sort back when we tested all this when @poire-z revamped it. Important timeline data point: I didn't have a Clara 2E at the time, and that device uses a panel (fts, from Focaltech) I've never seen anywhere else, AFAIR.

So, I finally had time to looks at the evdev stream in situ, and... things got weird. On the upside, the event stream is pretty sane, none of that snow/phoenix nonsense.
And when you look at a slow pan across the screen, at first things look *mostly* okay... as long as the device is idle!
But in the PDF reader, the device is very much *not* idle when panning ;).
And that's when shit starts to hit the fan: you'll start to see *actual* contact lifts mid-stream, and that sometimes trips a spurious *second* slot to be detected, while in fact, there's only a single contact involved.

It *seems* to be more reliable when the contact area is larger (i.e., with the pad of a finger instead of the tip), but that's not a silver bullet, either.

TL:DR: Not our bug, the driver is *actually* reporting spurious input events.

NOTE: I don't recall if that was the device in question, but *busy* here might specifically mean "the EPDC being busy", as I recall touch panel woes from the Quill guys when working with the mainline kernel on a specific panel/device, where it was very upset about interference from *something*, and that something was probably the EPDC/eInk panel.

#### Steps to reproduce

Single contact slow pan on a busy (see note above) device.

It's possibly easier to follow in `evemu_record`, as you'll notice the `ABS_MT_TRACKING_ID` incrementing multiple times when things go wrong (and possibly an `ABS_MT_SLOT 1` popping up when shits goes *really* wrong).

**Comment by NiLuJe** [comment]
To offset my snarky title, I'm otherwise pretty happy with the device, and swipe scrolling (i.e., "on release scrolling") works just fine ;).

**Comment by mergen3107** [comment]
@NiLuJe 
Do you know if Kobo Clara BW is prone to the same thing?

**Comment by NiLuJe** [comment]
I don't remember which panel that device is using ;). I'll try to find a
hwconfig dump to check.

**Comment by NiLuJe** [comment]
> Do you know if Kobo Clara BW is prone to the same thing?

It shouldn't, it's using a more common CYTT21X (Cypress) panel.

**Comment by mergen3107** [comment]
Hooray, thanks! 
I am really considering to get one soon :D

**Comment by ummm288** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.
